### PR TITLE
add documentation for handling failures

### PR DESCRIPTION
**Pull request by furkansenharputlu**
This PR adds documentation for handling failures. #340

### CODE REVIEW

**Comment by furkansenharputlu** [comment]
verify

**Comment by furkansenharputlu** [comment]
verify

File: README.md [hunk 1]
```diff
@@ -1234,6 +1237,33 @@ As a final step, if you are done with your client, you can shut it down as shown
...
+
+- Member throws an exception.
+
+- Connection between the client and member is closed.
+
+- Client’s heartbeat requests are timed out.
+
+When a connection problem occurs, an operation is retried if it is certain that it has not run on the member yet or if it is idempotent such as a read-only operation, i.e., retrying does not have a side effect. If it is not certain whether the operation has run on the member, then the non-idempotent operations are not retried. However, as explained in the first paragraph of this section, you can force all client operations to be retried (`redoOperation`) when there is a connection failure between the client and member. But in this case, you should know that some operations may run multiple times causing conflicts. For example, assume that your client sent a `queue.offer` operation to the member, and then the connection is lost. Since there will be no respond for this operation, you will not now whether it has run on the member or not. If you enabled `redoOperation`, it means this operation may run again, which may cause two sane objects in the queue.
```
Comment (mustafaiman) [inline README.md]: `respond` -> `response`

Comment (mustafaiman) [inline README.md]: `two sane objects` -> `two instances of the same object`

**Pull request merged**